conf/modules.config

```
@@ -59,13 +85,47 @@ process {
         cpus   = { check_max( 2     * task.attempt, 'cpus'    ) }
         memory = { check_max( 12.GB * task.attempt, 'memory'  ) }
         time   = { check_max( 4.h   * task.attempt, 'time'    ) }
+        ext.prefix = "decontaminated"
     }
 
     withName: 'SPADES' {
         ext.args = params.spades_only_assembler ? "--only-assembler" : ""
 
         memory = { "$params.assembly_memory" + ".GB" * task.attempt }
         cpus   = 32
+        publishDir = [
+            [
+                path: "${params.outdir}/assembly/spades/${params.spades_version}",
+                mode: params.publish_dir_mode,
+                failOnError: true,
+                pattern: "params.txt*"
+
+            ],
+            [
+                path: "${params.outdir}/assembly/spades/${params.spades_version}",
+                mode: params.publish_dir_mode,
+                failOnError: true,
+                pattern: "*graph.fastg*"
+            ],
+            [
+                path: "${params.outdir}/assembly/spades/${params.spades_version}",
+                mode: params.publish_dir_mode,
+                failOnError: true,
+                pattern: "*contigs.fa.gz"
+            ],
+            [
+                path: "${params.outdir}/assembly/spades/${params.spades_version}",
+                mode: params.publish_dir_mode,
+                failOnError: true,
+                pattern: "*scaffolds.fa.gz"
+            ],
+            [
+                path: "${params.outdir}/assembly/spades/${params.spades_version}",
+                mode: params.publish_dir_mode,
+                failOnError: true,
+                pattern: "*graph_with_scaffolds.gfa.gz"
+            ]
```

I think this should work (less repetition)
```suggestion
            [
                path: "${params.outdir}/assembly/spades/${params.spades_version}",
                mode: params.publish_dir_mode,
                failOnError: true,
                saveAs: {
                    if ( filename.equals('versions.yml') ) {
                        return null;
                    }
                    def file_patterns_to_publish = [
                        "params.txt",
                        "graph.fastg",
                        "contigs.fa.gz",
                        "scaffolds.fa.gz",
                        "graph_with_scaffolds.gfa.gz"
                    ]
                    if ( file_patterns_to_publish.any { it.contains(filename) } ) {
                        return filename;
                    }
                    return null;
                }
            ]
```